tools/images/ubuntu/Dockerfile

```diff
@@ -2,29 +2,26 @@ FROM ubuntu:xenial
 MAINTAINER peter.ebden@gmail.com
 
 # Basic dependencies, Python and Java
-RUN apt-get update && apt-get install -y python2.7 python3.5 python-pip python3-pip openjdk-8-jdk curl unzip git locales
-RUN pip install cffi
+RUN apt-get update
+RUN apt-get install -y python2.7 python3.5 python3-pip libpython2.7-dev openjdk-8-jdk-headless curl unzip git locales pkg-config zlib1g-dev
```

Comment:
the `apt-get update` should be on the same line as the `apt-get install` or you'll have docker layer caching issues

Reply:
Done, I've collapsed all the apt stuff into one step to save on work.